tests/functional/aws-node-sdk/test/bucket/getWebsite.js

```diff
@@ -0,0 +1,94 @@
+import assert from 'assert';
+
+import withV4 from '../support/withV4';
+import BucketUtility from '../../lib/utility/bucket-util';
+import { WebsiteConfigTester } from '../../lib/utility/website-util';
+
+const bucketName = 'testgetwebsitebucket';
+const ruleRedirect1 = {
+    HostName: 'test',
+    ReplaceKeyPrefixWith: 'documents/',
+};
+const ruleCondition1 = {
+    KeyPrefixEquals: 'docs/',
+};
+const ruleRedirect2 = {
+    HttpRedirectCode: 302,
+};
+const ruleCondition2 = {
+    HttpErrorCodeReturnedEquals: 404,
+};
+const config = new WebsiteConfigTester('index.html', 'error.html');
+config.addRoutingRule(ruleRedirect1, ruleCondition1);
+config.addRoutingRule(ruleRedirect2, ruleCondition2);
+
+describe('GET bucket website', () => {
+    withV4(sigCfg => {
+        const bucketUtil = new BucketUtility('default', sigCfg);
+        const s3 = bucketUtil.s3;
+
+        afterEach(() => {
+            process.stdout.write('about to delete bucket\n');
+            return bucketUtil.deleteOne(bucketName)
+            .catch(err => {
+                if (err) {
+                    process.stdout.write('error in afterEach', err);
+                    throw err;
+                }
+            });
+        });
+
+        describe('with existing bucket configuration', () => {
+            before(done => {
+                process.stdout.write('about to create bucket\n');
+                s3.createBucket({ Bucket: bucketName }, err => {
+                    if (err) {
+                        process.stdout.write('error creating bucket', err);
+                        return done(err);
+                    }
+                    process.stdout.write('about to put bucket website\n');
+                    s3.putBucketWebsite({ Bucket: bucketName,
+                        WebsiteConfiguration: config }, err => {
+                        if (err) {
+                            process.stdout.write('error putting bucket website',
+                                err);
+                            return done(err);
+                        }
+                        return done();
+                    });
+                    return undefined;
+                });
+            });
+
+            it('should return bucket website xml successfully', done => {
+                s3.getBucketWebsite({ Bucket: bucketName }, err => {
+                    assert.strictEqual(err, null,
+                        `Found unexpected err ${err}`);
+                    return done();
```

Comment:
~~How about asserting the xml returned to make sure mandatory elements are always set?~~
Edit: I see the tests in the other file. How about combing these two?